docs/se/faulttolerance/01_faulttolerance.adoc

```diff
@@ -0,0 +1,338 @@
+
+[source,java]
+----
+Single<Thread> s = Async.create().invoke(() -> Thread.currentThread()));
+s.thenAccept(t -> System.out.println("Async task executed in thread " + t));
+----
+
+The supplier `() -> Thread.currentThread()` is executed in a new thread and
+the value it produces printed by the consumer passed to `thenAccept`.
+
+NOTE: The method reference `Thread::currentThread` is a simplified way of
+providing a supplier in the example above.
+
+Asynchronous tasks are executed in a thread pool managed by the Helidon SE
+Fault Tolerance module. Thread pools are created during the initialization
+phase of class `io.helidon.faulttolerance.FaultTolerance` and can be
+configured for your application.
+
+=== Retries
+
+Temporal networking problems can sometimes be mitigated by simply retrying
+a certain task. A `Retry` handler is created using a `RetryPolicy` that
+indicates the number of retries, delay between retries, etc.
+
+[source,java]
+----
+Retry retry = Retry.builder()
+                   .retryPolicy(Retry.JitterRetryPolicy.builder()
+                                     .calls(3)
+                                     .delay(Duration.ofMillis(100))
+                                     .build())
+                   .build();
+retry.invoke(this::retryOnFailure);
+----
+
+The sample code above will retry calls to the supplier `this::retryOnFailure`
+for up to 3 times with a 100 millisecond delay between them.
+
+NOTE: The return type of method `retryOnFailure` in the example above must
+be `CompletionStage<T>` and the parameter to the retry handler's `invoke`
+method `Supplier<? extends CompletionStage<T>>`.
+
+If the `CompletionStage<T>` returned by the method completes exceptionally,
+the call will be treated as a failure and retried until the maximum number
+of attempts is reached; finer control is
+possible by creating a retry policy and using methods such as
+`applyOn(Class<? extends Throwable>... classes)` and
+`skipOn(Class<? extends Throwable>... classes)` to control those exceptions
+on which to act and those that can be ignored.
+
+=== Timeouts
+
+A request to a service that is inaccessible or simply unavailable should be bounded
+to ensure a certain quality of service and response time. Timeouts can be configured
+to avoid excessive waiting times. In addition, a fallback action can be defined
+if a timeout expires as we shall cover in the next section.
+
+The following is an example of using `Timeout`:
+[source,java]
+----
+Single<T> s = Timeout.create(Duration.ofMillis(10)).invoke(this::mayTakeVeryLong);
+s.handle((t, e) -> {
+    if (e instanceof TimeoutException) {
+        // Invocation has timed out!
+    }
+    ...
+});
+----
+
+The example above monitors the call to method `mayTakeVeryLong` and reports a
+`TimeoutException` if the execution takes more than 10 milliseconds to complete.
+
+=== Fallbacks
+
+A fall back to a _known_ result can sometimes be an alternative to
```

Comment:
A fallback to a _known_ result can ...

fallback is the noun form and fall back is the action